sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -802,8 +752,11 @@ def __eq__(self, other: Any) -> bool:
             return False
         if self._trim_prefixes != other._trim_prefixes:
             return False
-        if self._client != other._client:
+        if len(self._replica_client_manager._replica_clients) != len(other._replica_client_manager._replica_clients):
```

Comment:
Maybe the  equality logic for replica clients could move to the replica client manager?